contracts/blender/src/contract.rs

```diff
@@ -51,6 +51,14 @@ mod blender {
     type Result<T> = core::result::Result<T, BlenderError>;
     type Event = <Blender as ContractEventBase>::Type;
 
+    /// Describes a path from a leaf to the root.
+    ///
+    /// It is represented as sibling hashes list (bottom to top). The pairs are always of form
+    /// `(left hash, right hash)` with the exception of the last pair, which is just
+    /// `(root hash, root hash)`. The requested leaf is put in the first pair (check index parity
+    /// to find out the coordinate).
+    pub type MerklePath = Vec<(Hash, Hash)>;
```

Comment:
It should be possible to represent this as just `Vec<Hash>` since leaves/nodes are at deterministic positions:
* head will be our starting leaf at `leaf_idx`
* second element would be its sibiling
* third element is sibiling of their parent 
* fourth element sibiling of the previous' elements parent
* etc.

I also wouldn't send the `root_hash` as it's not necessarily required - it is recoverable from the merkle path and the leaf.

And for root computation we would [`reduce`](https://doc.rust-lang.org/nightly/core/iter/trait.Iterator.html#method.reduce) over the collection with `f` being the hashing function.

Or am I wrong again?

Reply:
generally fair enough - I will flatten this struct

two notes however:
 - head will be the 'lefter' of our leaf and its sibling
 - second element will be `righter` of our leaf and its sibling
 - third element will be `lefter` of their parent and its sibling
 - fourth element will be `righter` of their parent and its sibling
 
and `reduce` would truly be cute and elegant, but eventually we won't use it (this list, maybe somehow transformed, will be passed to the circuit and there Jedrzej will use some ready code from arkworks)

Reply:
done

Reply:
🙈